js/viz/core/renderers/renderer.js

```diff
@@ -1987,6 +2007,31 @@ Renderer.prototype = {
         return pattern;
     },
 
+    customPattern: function(id, templateFunction, width, height) {
```

Comment:
It's not necessary to specify variable's type in its name. `template` is meaningful enough
```suggestion
    customPattern: function(id, template, width, height) {
```